src/reflect/value.go

```diff
@@ -980,7 +980,13 @@ func (v Value) SetString(x string) {
 }
 
 func (v Value) SetBytes(x []byte) {
-	panic("unimplemented: (reflect.Value).SetBytes()")
+	v.checkAddressable()
+	if v.typecode.Kind() != Slice || v.typecode.elem().Kind() != Uint8 {
+		panic("reflect.Value.SetBytes called on not []byte")
+	}
+
+	// copy the header contents over
+	*(*sliceHeader)(v.value) = *(*sliceHeader)(unsafe.Pointer(&x))
```

Comment:
I think this works just as well, and seems easier to read:

```suggestion
	*(*[]byte)(v.value) = x
```

Reply:
Oh yeah, of course.